composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/domain/ValidateEmailUseCase.kt

```diff
@@ -0,0 +1,12 @@
+package sp.bvantur.tasky.core.domain
+
+interface ValidateEmailUseCase {
+    operator fun invoke(email: String): Boolean
+}
+
+class ValidateEmailUseCaseImpl : ValidateEmailUseCase {
```

Comment:
💯

Reply:
I am removing the abstraction here. This is a good point. I act here from my development habits with KMP projects because I was usually also testing ViewModel classes, and AFAIK as of now there is no testing library that provides you with a mocking of class directly. You always need to abstract them with an interface to make that work. For this project, I will not write tests for ViewModel so we don't need this. If any of you know of a way how to mock classes on KMP projects I would love to learn more about that.

Reply:
From my personal testing experience, those tests that are truly useful are those that test multiple parts of the app as a whole (integration and end-to-end tests). Not to say that unit tests are useless - they are, especially if they're quick to write, but I never saw a lot of value in abstracting use cases that purely contain business logic. So it's not about passing in a test double for the sake of removing a network or DB source, but just to insist on isolating the ViewModel.
But think about it, what's the worst that could happen if you wouldn't isolate the ViewModel by passing in use case test doubles (and just passing in the real use cases instead)? The worst is that your test fails because of an issue in one of these use cases, but hints towards the ViewModel. Is this one little problem (if it can be considered a problem at all) really worth all that abstraction effort and making the codebase so much more difficult to navigate and understand? I don't think so.